plog-client/src/main/java/com/airbnb/plog/client/PlogClient.java

```diff
@@ -0,0 +1,156 @@
+package com.airbnb.plog.client;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+import com.google.common.hash.Hashing;
+
+/**
+ * ## Plog client for Java
+ * Example:
+ *
+ * ```java
+ * PlogClient plogClient = new PlogClient("127.0.0.1", 23456, 64000);
+ * plogClient.send("My hovercraft is full of eels.");
+ * ```
+ * You can configure the client at initialization by passing these options:
+ *
+ * + host - The host of the Plog process (e.g., 'localhost')
+ * + port - The port on which Plog is listening (e.g., 23456)
+ * + chunkSize - The maximum payload size for multipart datagrams (e.g., 64,000)
+ */
+@Slf4j
+public class PlogClient implements Closeable {
+
+  public static final int DEFAULT_CHUNK_SIZE = 64000;
+
+  private AtomicInteger lastMessageId;
+
+  private final InetAddress address;
+
+  private final int port;
+
+  private final int chunkSize;
+
+  private DatagramSocket socket;
+
+  public PlogClient(String host, int port) {
+    this(host, port, DEFAULT_CHUNK_SIZE);
+  }
+
+  public PlogClient(String host, int port, int chunkSize) {
+    Preconditions.checkNotNull(host, "host cannot be null!");
+    Preconditions.checkArgument(port > 1024 && port < 65536, "Must provide a valid port number!");
+    Preconditions.checkArgument(chunkSize < 65507, "Maximum UDP data length is 65507 bytes!");
```

Comment:
will a chunk size of `65507` be ok? This value does not take the Plog header size into account